src/workflow-handler.ts

```diff
@@ -127,64 +130,47 @@ export class WorkflowHandler {
       return this.workflowRunId;
     }
     try {
-      core.debug('Get workflow run id');
+      const workflowId = await this.getWorkflowId();
+
+      const response = await this.octokit.rest.actions.listWorkflowRuns({
+        owner: this.owner,
+        repo: this.repo,
+        workflow_id: workflowId,
+        event: 'workflow_dispatch',
+        created: `>=${new Date(this.triggerDate).toISOString()}`
+      });
+
+      debug('List Workflow Runs', response);
+
+      let runs = response.data.workflow_runs;
       if (this.runName) {
-        this.workflowRunId = await this.findWorklowRunIdFromRunName(this.runName);
-      } else {
-        this.workflowRunId = await this.findWorkflowRunIdFromFirstRunOfSameWorkflowId();
+        runs = response.data.workflow_runs
+          .filter((r: any) => r.name == this.runName)
+      }
+
+      if (runs.length == 0) {
+        throw new Error('Run not found');
+      }
+
+      if (runs.length > 1) {
+        core.warning(`Found ${runs.length} runs. Using the last one.`);
+        debug(`Filtered Workflow Runs (after trigger date: ${new Date(this.triggerDate).toISOString()})`, runs.map((r: any) => ({
```

Comment:
Same here, too much code for a simple debug, extract in a method for formatting message